nengo_gui/components/pointer.py

```diff
@@ -1,32 +1,48 @@
-import collections
 import copy
 
 import nengo
-import nengo.spa
-from nengo.spa.module import Module
+import nengo.spa as spa
 import numpy as np
 
 from nengo_gui.components.component import Component
 from nengo_gui.components.spa_plot import SpaPlot
 
+
 class Pointer(SpaPlot):
     config_defaults = dict(show_pairs=False, **Component.config_defaults)
     def __init__(self, obj, **kwargs):
         super(Pointer, self).__init__(obj, **kwargs)
+
         # the semantic pointer value as set by the user in the GUI
         # a value of 'None' means do not override
         self.override_target = None
-        self.vocab_in = obj.inputs[self.target][1]
+
+        # The white list indicates the networks whose user-defined
+        # over-ride value can be inserted on the input
+        # thus the value loops in from the output to the input.
+        # All other networks have their value inserted on the output.
+        # Looping-in has the advantage of actually changing the
+        # neural activity of the population, rather than just changing
+        # the output.
+        self.loop_in_whitelist = [spa.Buffer, spa.Memory, spa.State]
+
+        self.node = None
+        self.conn1 = None
+        self.conn2 = None
 
     def add_nengo_objects(self, page):
         with page.model:
             output = self.obj.outputs[self.target][0]
-            input = self.obj.inputs[self.target][0]
             self.node = nengo.Node(self.gather_data,
                                    size_in=self.vocab_out.dimensions,
-                                   size_out=self.vocab_in.dimensions)
+                                   size_out=self.vocab_out.dimensions)
             self.conn1 = nengo.Connection(output, self.node, synapse=0.01)
-            self.conn2 = nengo.Connection(self.node, input, synapse=0.01)
+            if (type(self.obj) in self.loop_in_whitelist
+                and self.target == 'default'):
```

Comment:
This line does not distinguish itself from the indented line below, making it very hard to parse.

Note that this is exactly the one annoying case in Python indenting (an if statement with a parenthesis continuation, meaning that the natural amount you'd want to indent the second part of the first line is exactly four spaces).  I normally just rewrite the logic a bit to avoid having to do the line continuation.